src/wwwroot/app/directives/event-calendar/event-calendar.js

```diff
@@ -95,45 +120,71 @@ function EventCalendarController($scope) {
          vm.currentYear = vm.currentYear - 1;
          vm.currentMonth = 11;
       }
+      formattingDataToCalendarView();
+   }
+
+   function goToNextYear() {
+      vm.currentYear = vm.currentYear + 1;
+      formattingDataToCalendarView();
+   }
+
+   function goToPreviousYear() {
+      vm.currentYear = vm.currentYear - 1;
+      formattingDataToCalendarView();
+   }
+
+   function formattingDataToCalendarView() {
       vm.daysInMonth = [];
       vm.eventsForMonth  = {};
       vm.dateForSearchEvents = new Date(vm.currentYear, vm.currentMonth);
-      getDaysInMonth(vm.currentMonth, vm.currentYear);
-      vm.getEvents(new Date(vm.currentYear, vm.currentMonth, 1), vm.checkedUsers).then((events) => {
+      formatingSearchCondition();
+      getDaysInMonth(vm.startDate, vm.endDate);
+      vm.getEvents(vm.startDate, vm.endDate, vm.checkedUsers).then((events) => {
```

Comment:
you can call method in 'then' instead of creating a new clause. like 
**vm.getEvents(...).then(convertEventsToHash);**